### ISSUE TITLE
Re-implement task system

### ISSUE THREAD

**Issue by nicemicro** [opensuspect/opensuspect#39]
- [x] 1. Contains the back end for saving and syncing task state.

- [x] 2. Interaction with tasks is possible.

- [x] 3. Tasks can have effect on the game state / other tasks (i.e. gas pressure "maintenance" task)

- [x] 4. Two task prototypes implemented.

- [x] 5. In this implementations, no task assignment needed, all tasks should act as global that anyone can do.

**Comment by monban** [comment]
I've been thinking about the task system and how it might be implemented.

Object | Purpose
---------|------------
TaskList | Basically an array of TaskDetails objects
TaskDetails | The implementation of of a specific task. What objects are required, conditions to make progress, complete
Map | Contains a list of possible tasks (from the Task Data) and the Task Objects required to complete them
TaskManager | Creates Tasks by requesting corresponding TaskDetails from the Map's list, and assigns them to Players, tracks completion status, sends updates
TaskObject | Can be interacted with to complete tasks (by communicating with Task Manager)
Task | Contains a list of requirements for completion (by linking to a TaskDetails) and tracks completion
Player | Completes Tasks by interacting with Task Objects

```mermaid
sequenceDiagram
    Note right of TaskManager: Get possible tasks from Map
    Note right of TaskManager: Get TaskDetails from TaskList
    Note right of TaskManager: Create list of Tasks
    TaskManager->>+Player: Here are your assigned tasks
    Player->>+TaskObject: Interact
    TaskObject->>+TaskManager: Task completion progression
    TaskManager->>+Player: Update task lists
```

This structure allows us to define Tasks separate from each Map, each Map creator can choose from that list of tasks to implement. This prevents us from needing to redefine the same tasks on multiple maps. Does this make sense?

**Comment by nicemicro** [comment]
Makes sense, @monban .
Based on my experience with the development of the role system, and what I learned about resources from @TheSecondReal0 , I'd say, that the TaskObjects should be resources, which can handle special behavior and store data, but each task should be always in communication with TaskNodes that are instanced on the map (probably not from code, but from the editor, to make map editing more easy) and also TaskUI, which should be opened whenever a task is active.
I.e., a chemical cabinet could have either closed or opened doors, and that should be reflected for everyone on the map, and for the person active at the task, that should be seen on the task screen.

**Comment by nicemicro** [comment]
All implemented by #102
